### ISSUE TITLE
Support building more than one document

### ISSUE THREAD

**Issue by arm61** [showyourwork/showyourwork#287]
I would like to build my paper and a separate supplementary information document both from LaTeX sources. I imagine I could do this with some additional Snakefile content but was wondering if there was a smarter way.

**Comment by dfm** [comment]
Great question! Would you like both documents to be showyourwork'ed? For example, are there figure scripts in both and do you want the margin icons in both? I don't think that's currently supported and might take some hacking, but this would be a great use case (that we hadn't thought of before) to support!

**Comment by arm61** [comment]
Yeah I both would be showyourwork'd. What do the astronomy community usually do for SI?

**Comment by dfm** [comment]
Got it - our journals normally don't have a "supplementary information" document separately, so it's not something we typically support. I'll think a little more about how that would fit into the current structure. (It won't be easy, I don't think, but we might be able to come up with something.)

It'll be easier with https://github.com/showyourwork/showyourwork/pull/285, but that's probably a still ways off from production ready...

**Comment by arm61** [comment]
There isn't a rush on this. For arXiv submission, I plan to submit as a single document and the journal submission I am planning to remove most of the showyourwork stuff (😢) cause we wanna submit to APS and this is a bit off putting https://github.com/showyourwork/showyourwork/issues/280.

**Comment by dfm** [comment]
I've come up with a scheme for supporting this post refactor in #285, but I think it's probably infeasible in the existing infrastructure. The idea that there is a single manuscript is very tightly integrated into the SYW worldview, and I expect it would be fairly painful to break that.

I'm going to keep this open as a design requirement for the next version, but let's not stress it too much for the short term if that's ok with you?

**Comment by HealthyPear** [comment]
Another use case for this would be to couple a paper to one or more presentations / posters.

**Comment by HealthyPear** [comment]
Regarding this issue I saw that basically there are at least two places where the manuscript is pinned for compilation:

- in `src/showyourwork/workflow/rules/compile.smk`
  -  it is defined by some keys in `showyourwork.yml` (`config["ms_tex"]` and `config["dependencies"][config["ms_tex"]]`)
  -  the files themselves are defined by the input function `temporary_tex_files()`

`tectonic` and its options should not likely change between the main manuscript and, say, a beamer presentation.

the temporary files are themselves defined by a rule named "syw__compile_setup" which again depends on the yaml keys mentioned above.

What would happen if we call the compile workflow by injecting the name of the key related to the document to compile?
